### DISCUSSION TITLE
<style> order problem in Generated Template

### DISCUSSION THREAD

**Discussion by Xtreemrus** [zauberzeug/nicegui#2353] [Ideas / Feature Requests]
Hello everyone,

I've noticed that in the generated template, the style added through `add_head_html` comes first, followed by the standard Tailwind style below.

This order leads to the user-added CSS properties for standard elements being overwritten by Tailwind.

If we switch the order, the rules reassigned by the user should become the priority.

Here's an example of how custom CSS is added:

```python
@ui.page('/')
async def main(client: Client):
    
    ui.add_head_html('''
     <style>
        <!-- my custom css -->
               
        h2 {
            font-size: 1.875rem;
            line-height: 2.25rem;
            font-weight: 700;
        }
        
     
    </style>
    ''')
```

I've also attached a screenshot for clarity.

![NiceGUI_change_style order](https://github.com/zauberzeug/nicegui/assets/34982372/92981d57-f822-46ef-bccd-d5eb87dbc500)

I believe adjusting the order in which styles are applied could enhance the customization experience.

**Comment by falkoschindler** [comment]
Thanks for the suggestion, @Xtreemrus!

I'm just not sure how to achieve this, because Tailwind's CSS definitions are inserted via JavaScript at runtime. In the HTML template provided by the server, `head_html` is indeed the last line in \<head\>:

https://github.com/zauberzeug/nicegui/blob/59c3dee195331f4a4c7c54eeb45abe05fc0af0a1/nicegui/templates/index.html#L1-L22

**Reply by Xtreemrus** [reply]
Hooray, thanks to your answer to a similar question, a solution has been found here as well.
I think the most general recommendation is that if a launch is made with the parameter tailwind=True (ui.run(tailwind=True)), it's better to define styles with the type attribute "text/tailwindcss", so that the styles defined by the developer are not overwritten by tailwind, and the final say always remains with the developer.

```py
@ui.page('/')
async def main(client: Client):
    
    ui.add_head_html('''
     <style type="text/tailwindcss">
        <!-- my custom css -->
               
        h2 {
            font-size: 1.875rem;
            line-height: 2.25rem;
            font-weight: 700;
        }
        
     
    </style>
    ''')
```

Perhaps mentioning this in the NiceGUI style documentation could help current and future users who will start using it to create landing pages :)

**Reply by falkoschindler** [reply]
Oh wow, that's a great observation! We should definitely mention it in the documentation. 👍🏻

**Reply by falkoschindler** [reply]
I just added a demo to the documentation: https://github.com/zauberzeug/nicegui/commit/d06a472cd75e7441aa3a073e3c88972d63a228b3
